packages/grid_client/tests/modules/gateways.test.ts

```diff
@@ -190,15 +192,34 @@ test("TC1237 - Gateways: Expose a VM Over Gateway", async () => {
   expect(gatewayResult[0].backends).toStrictEqual(backends);
 
   const domain = "https://" + gatewayResult[0].domain;
+  let reachable = false;
+
+  for (let i = 0; i < 15; i++) {
+    fetch(domain, { mode: "no-cors" })
+      .then(r => {
+        log("gateway is reachable");
+        reachable = true;
+        i = 15;
```

Comment:
why setting `i = 15`? Is it to break the loop?

Reply:
yes it's for breaking the loop. but i'll check if there's a cleaner way to do it

Reply:
you can use `break`

Reply:
done in https://github.com/threefoldtech/tfgrid-sdk-ts/pull/107/commits/de767e12e94a87712fd099f88828ca329928ff9d i was unable to use `break` inside the fetch method so that what made me use the `i=15` at first but i found a better way of using `break` but outside the fetch method with a condition using the `reachable` variable.